### PR TITLE
fix: [#185940232] differentiate foreign and domestic legal type labels for formation

### PR DESCRIPTION
**Pull request by aloverso**
### Ticket

[185940232](https://www.pivotaltracker.com/story/show/#185940232)

## Code author checklist

- [ ] I have performed a self-review of my code
- [ ] I have created and/or updated relevant documentation, if necessary
- [ ] I have not used any relative imports
- [ ] I have checked for and removed instances of unused config from CMS
- [ ] I have pruned any instances of unused code
- [ ] I have not added any markdown to labels, titles and button text in config
- [ ] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [ ] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts`

### CODE REVIEW

File: web/src/components/tasks/business-formation/business/BusinessStep.test.tsx [hunk 1]
```diff
@@ -812,44 +812,62 @@ describe("Formation - BusinessStep", () => {
       const legalStructureId = "limited-liability-company";
       await getPageHelper({ legalStructureId }, {});
```
Comment (adidner) [inline web/src/components/tasks/business-formation/business/BusinessStep.test.tsx:813]: [sand] These tests are really repetitive and could probably use an it.each to iterate through instead

Comment (adidner) [reply web/src/components/tasks/business-formation/business/BusinessStep.test.tsx:813]: [sand] also a bit should we also test for foreign in the same fashion here?

File: web/src/components/tasks/business-formation/business/BusinessStep.test.tsx [hunk 2]
```diff
@@ -812,44 +812,62 @@ describe("Formation - BusinessStep", () => {
...
       const displayLegalStructure = screen.getByTestId("legal-structure");
-      expect(displayLegalStructure).toHaveTextContent(Config.formation.legalStructure[legalStructureId]);
+      expect(displayLegalStructure).toHaveTextContent(
+        Config.formation.legalStructure.domesticLabels[legalStructureId]
+      );
     });
 
-    it("displays foreign label ahead of legal structure from profile data", async () => {
+    it("displays foreign label for legal structure for FOREIGN persona", async () => {
       const legalStructureId = randomPublicFilingLegalType();
```
Comment (adidner) [inline web/src/components/tasks/business-formation/business/BusinessStep.test.tsx:857]: [dust] might be good to have a refactor ticket for gradually replacing random functions with it.each function for robustness. I don't think it would add much time to the CI tests (none in fact because I think the cypress stuff is still the limiting reactant) and I think coverage for tests in this regard is always good.

**Review by adidner — APPROVED** [review]
Minor comments to potentially make tests a bit cleaner and potential follow up refactor.

File: web/src/components/tasks/business-formation/business/BusinessNameAndLegalStructure.tsx [hunk 3]
```diff
@@ -45,14 +45,14 @@ export const BusinessNameAndLegalStructure = ({ isReviewStep = false }: Props):
...
-      business?.profileData.businessPersona === "FOREIGN"
-        ? `${Config.formation.legalStructure.foreignPrefaceText} `
-        : "";
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const legalStructure = (Config.formation.legalStructure as any)[business.profileData.legalStructureId];
-    return `${preface} ${legalStructure}`;
+    if (business.profileData.businessPersona === "FOREIGN") {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      return (Config.formation.legalStructure.foreignLabels as any)[business.profileData.legalStructureId];
```
Comment (jphechter) [inline web/src/components/tasks/business-formation/business/BusinessNameAndLegalStructure.tsx:51]: It'd be cool if we didn't need `as any`, but I don't know if that's worth the effort. This is fine.
